ui/src/shared/components/TimeMachine/RawFluxDataTable.tsx

```diff
@@ -48,24 +52,53 @@ class RawFluxDataTable extends PureComponent<Props, State> {
               MIN_COLUMN_WIDTH,
               resolvedWidth / maxColumnCount
             )
+            const gridScrollWidth = columnWidth * maxColumnCount
+            const gridScrollHeight = ROW_HEIGHT * data.length
 
             return (
-              <Grid
-                width={resolvedWidth}
-                height={resolvedHeight}
-                cellRenderer={this.renderCell}
-                columnCount={maxColumnCount}
-                rowCount={data.length}
-                rowHeight={ROW_HEIGHT}
-                columnWidth={columnWidth}
-              />
+              <FancyScrollbar
+                style={{
+                  overflowY: 'hidden',
+                  width: resolvedWidth,
+                  height: resolvedHeight,
+                }}
+                autoHide={false}
+                scrollTop={scrollTop}
+                scrollLeft={scrollLeft}
+                setScrollTop={this.onScrollbarsScroll}
+              >
+                <Grid
+                  width={resolvedWidth}
+                  height={resolvedHeight}
+                  cellRenderer={this.renderCell}
+                  columnCount={maxColumnCount}
+                  rowCount={data.length}
+                  rowHeight={ROW_HEIGHT}
+                  columnWidth={columnWidth}
+                  scrollLeft={scrollLeft}
+                  scrollTop={scrollTop}
+                  style={{
+                    width: gridScrollWidth,
+                    height: gridScrollHeight,
+                  }}
+                />
+              </FancyScrollbar>
             )
           }}
         </AutoSizer>
       </div>
     )
   }
 
+  private onScrollbarsScroll = e => {
```

Comment:
If `FancyScrollBar`'s onScrollTop was typed with the proper generic you probably wouldn't have to coerce the target to an HTMLElement